File: snippets/fsharp/System/ValueType/ToString/ToString2.fs
// <Snippet1>
namespace Corporate.EmployeeObjects

[<Struct>]
type EmployeeA =
    val mutable Name : string

[<Struct>]
type EmployeeB =
    val mutable Name : string
    override this.ToString() = 
          this.Name

module Example =
     let empA = EmployeeA(Name="Robert")
     printfn $"{empA}"

     let empB = EmployeeB(Name="Robert")
     printfn $"{empB}"
// The example displays the following output:
//     Corporate.EmployeeObjects.EmployeeA
//     Robert
// </Snippet1>

File: snippets/csharp/System/ValueType/ToString/ToString2.cs
// <Snippet1>
using System;
using Corporate.EmployeeObjects;

public class Example
{
    public static void Main()
    {
        var empA = new EmployeeA { Name = "Robert", };
        Console.WriteLine(empA.ToString());

        var empB = new EmployeeB { Name = "Robert", };
        Console.WriteLine(empB.ToString());
    }
}

namespace Corporate.EmployeeObjects
{
    public struct EmployeeA
    {
        public string Name { get; set; }
    }

    public struct EmployeeB
    {
        public string Name { get; set; }

        public override string ToString() => Name;
    }
}
// The example displays the following output:
//     Corporate.EmployeeObjects.EmployeeA
//     Robert
// </Snippet1>
